### PR TITLE
fix(lsp): show dependency errors for repeated imports

### PR DESCRIPTION
**Pull request by nayeemrmn**
This was missing because `deno_graph` didn't store every import site for a dependency until https://github.com/denoland/deno_graph/pull/251.

### Before
image

### After
image

### CODE REVIEW

**Comment by bartlomieju** [comment]
@nayeemrmn could you bump `deno_graph` dependency in this PR?

File: cli/lsp/diagnostics.rs [hunk 1]
```diff
@@ -1005,17 +996,43 @@ fn diagnose_dependency(
...
+  if !dependency.maybe_type.is_none()
+    && !dependency
+      .imports
+      .iter()
+      .any(|i| dependency.maybe_type.includes(&i.range.start).is_some())
+  {
+    let range = match &dependency.maybe_type {
+      Resolution::Ok(resolved) => documents::to_lsp_range(&resolved.range),
+      Resolution::Err(error) => documents::to_lsp_range(error.range()),
+      Resolution::None => unreachable!(),
```
Comment (dsherret) [inline cli/lsp/diagnostics.rs:1025]: Are we sure about this? Maybe just have it ignore this scenario instead?

File: cli/lsp/diagnostics.rs [hunk 2]
```diff
@@ -1376,4 +1393,64 @@ let c: number = "a";
...
+  #[tokio::test]
+  async fn duplicate_diagnostics_for_duplicate_imports() {
+    let temp_dir = TempDir::new();
+    let (snapshot, _) = setup(
+      &temp_dir,
+      &[(
+        "file:///a.ts",
+        r#"
+        import "bad.js";
+        import "bad.js";
```
Comment (dsherret) [inline cli/lsp/diagnostics.rs:1406]: Maybe try adding a test with multiple bad `@deno-types` directives as well?

**Review by dsherret — COMMENTED** [review]
Looks good. Just needs a few small changes.

File: cli/lsp/diagnostics.rs (hunk 1 above)
Comment (nayeemrmn) [reply cli/lsp/diagnostics.rs:1025]: It's gated in the containing if statement

**Review by dsherret — APPROVED** [review]
LGTM. Thanks @nayeemrmn!

File: cli/lsp/diagnostics.rs (hunk 2 above)
Comment (nayeemrmn) [reply cli/lsp/diagnostics.rs:1407]: Only one `@deno-types` directive is stored per JS dep, that would be addressed by https://github.com/denoland/deno_graph/pull/272.

**Pull request merged**